libkbfs/folder_branch_ops.go

```diff
@@ -230,9 +230,10 @@ type folderBranchOps struct {
 	// should only be taken in the following order to avoid deadlock:
 	mdWriterLock leveledMutex // taken by any method making MD modifications
 
-	// protects access to head and latestMergedRevision.
-	headLock leveledRWMutex
-	head     ImmutableRootMetadata
+	// protects access to head, headTrusted and latestMergedRevision.
+	headLock    leveledRWMutex
+	head        ImmutableRootMetadata
+	headTrusted bool
```

Comment:
I think trustedness is important enough to deserve its own type, say, `HeadTrustStatus` with values `HeadTrusted` and `HeadUntrusted`. It would also be a good place to put a comment about what exactly 'trusted' and 'untrusted' mean in this context.